code/src/reducers/Task.jsx

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable max-len */
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  items: [
+    {
+      id: 1,
+      text: 'Take out the trash',
+      isComplete: false
+    },
+    {
+      id: 2,
+      text: 'Buy milk',
+      isComplete: false
+    },
+    {
+      id: 3,
+      text: 'Never come back',
+      isComplete: false
+    }
+  ]
+};
+
+const Tasks = createSlice({
+  name: 'tasks',
+  initialState,
+  reducers: {
+    // first argument references the state of the store, second one the data passed down from the component
+    addTask: (store, action) => {
+      // Mutable
+      /* store.items.push(action.payload); */
+      // Immutable
+      store.items = [...store.items, action.payload];
+    },
+    deleteAllTasks: (store) => {
+      store.items = [];
+    },
+    deleteSingleTask: (store, action) => {
+      const id = action.payload;
+      // splice to remove a single element if I know the index
+      const copyOfTaskArrayFromStore = store.items;
+      const condition = (element) => element.id === id;
+      const foundIndex = copyOfTaskArrayFromStore.findIndex(condition);
+      copyOfTaskArrayFromStore.splice(foundIndex, 1);
+      store.items = copyOfTaskArrayFromStore;
+    },
+    toggleComplete: (store, action) => {
+      const id = action.payload;
+      const copyOfTaskArrayFromStore = store.items;
+      const condition = (element) => element.id === id;
+      const foundIndex = copyOfTaskArrayFromStore.findIndex(condition);
+      copyOfTaskArrayFromStore[foundIndex].isComplete = !copyOfTaskArrayFromStore[foundIndex].isComplete;
```

Comment:
That's cool! So you've added the foundIndex-function inside the toggle. I've seen people do this but never fully understood it. So the `[foundIndex]` in `copyOfTaskArrayFromStore[foundIndex].isComplete ` references the variable above, and is used as the index since it's between square brackets right? I did it like this: 

`toggleTask: (store, action) => {
      store.items.forEach((item) => {
        if (item.id === action.payload.taskId) {
          item.isDone = !item.isDone
        }
      })
    }`